libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp

```diff
@@ -168,40 +201,38 @@ LeddarOne_Status AP_RangeFinder_LeddarOne::send_request(void)
   */
 LeddarOne_Status AP_RangeFinder_LeddarOne::parse_response(uint8_t &number_detections)
 {
-    uint8_t data_buffer[25] = {0};
-    uint32_t start_ms = AP_HAL::millis();
-    uint32_t len = 0;
-    uint8_t i;
-    uint8_t index_offset = 11;
+    uint8_t index;
+    uint8_t index_offset = LEDDARONE_DATA_INDEX_OFFSET;
 
     // read serial
-    while (AP_HAL::millis() - start_ms < 10) {
-        uint32_t nbytes = uart->available();
-        if (len == 25 && nbytes == 0) {
-            break;
-        } else {
-            for (i=len; i<nbytes+len; i++) {
-                if (i >= 25) {
-                    return LEDDARONE_ERR_BAD_RESPONSE;
-                }
-                data_buffer[i] = uart->read();
+    uint32_t nbytes = uart->available();
+
+    if (nbytes != 0)  {
+        for (index=read_len; index<nbytes+read_len; index++) {
+            if (index >= 25) {
+                return LEDDARONE_ERR_BAD_RESPONSE;
             }
-            start_ms = AP_HAL::millis();
-            len += nbytes;
+            read_buffer[index] = uart->read();
+        }
+
+        read_len += nbytes;
+
+        if (read_len < 25) {
+            return LEDDARONE_READING_BUFFER;
         }
     }
 
```

Comment:
would be nice if the protocol packet was described somewhere, either here or top of function. Here's what I've gathered so far in an expected packet:
read_buffer[0] == ??? Sync?
read_buffer[1] == 0x04 = ??? Sync?? device type? mfgr?
read_buffer[2] == ???
read_buffer[3] == ???
...
read_buffer[10] == number of detections/reflections
read_buffer[11] == LEDDARONE_DATA_INDEX_OFFSET == first data index
...
read_buffer[12] == last index of data
read_buffer[23] = CRC.Low???
read_buffer[24] = CRC.High???